oauth2l-go/src/oauth2l/oauth2l.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import "flag"
+import "fmt"
+import "io/ioutil"
+import "oauth2client"
+
+func printUsage() {
+	fmt.Println("Usage: oauth2l --json <secret.json> {fetch|header|reset} " +
+		"scope1 scope2 ...")
+}
+
+func fetch(token map[string]interface{}) {
+	fmt.Println(token["access_token"])
+}
+
+func header(token map[string]interface{}) {
+	fmt.Printf("Authorization: Bearer %s\n", token["access_token"])
+}
+
+func main() {
+	json_file := flag.String("json", "", "Path to secret json file.")
+	help := flag.Bool("help", false, "Print help message.")
+	flag.BoolVar(help, "h", false, "")
+
+	flag.Parse()
+
+	if *help || len(flag.Args()) < 2 {
+		printUsage()
+		return
+	}
+
+	commands := map[string]func(map[string]interface{}){
+		"fetch":  fetch,
+		"header": header,
+	}
+	secretBytes, err := ioutil.ReadFile(*json_file)
+	if err != nil {
+		fmt.Printf("Failed to read file %s.\n", *json_file)
+		return
+	}
+
+	if cmdFunc, ok := commands[flag.Args()[0]]; ok {
```

Comment:
I prefer we use logic like this one:
  client := oauth2client.NewClient(secret)
  token := client.GetToken(scopes)

Let's copy the token type https://godoc.org/golang.org/x/oauth2#Token to here to avoid dependency.